src/sentry/interfaces/stacktrace.py

```diff
@@ -624,27 +626,28 @@ def __iter__(self):
         return iter(self.frames)
 
     @classmethod
-    def to_python(cls, data, has_system_frames=None, slim_frames=True):
+    def to_python(cls, data, has_system_frames=None, slim_frames=True,
+                  raw=False):
         if not data.get('frames'):
             raise InterfaceValidationError("No 'frames' present")
 
         if not isinstance(data['frames'], list):
             raise InterfaceValidationError("Invalid value for 'frames'")
 
-        if has_system_frames is None:
-            has_system_frames = cls.data_has_system_frames(data)
-
         frame_list = [
             # XXX(dcramer): handle PHP sending an empty array for a frame
-            Frame.to_python(f or {})
+            Frame.to_python(f or {}, raw=raw)
             for f in data['frames']
         ]
 
-        for frame in frame_list:
-            if not has_system_frames:
-                frame.in_app = False
-            elif frame.in_app is None:
-                frame.in_app = False
+        if not raw:
+            if has_system_frames is None:
+                has_system_frames = cls.data_has_system_frames(data)
+            for frame in frame_list:
+                if not has_system_frames:
+                    frame.in_app = False
+                elif frame.in_app is None:
+                    frame.in_app = False
 
         kwargs = {
             'frames': frame_list,
```

Comment:
Reading through this, it seems like the output of `to_python` is unaffected as long as `raw` is `False` (which is the default).